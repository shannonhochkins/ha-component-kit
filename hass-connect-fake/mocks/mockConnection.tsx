import type { HassEntities } from "home-assistant-js-websocket";
import 'jest-localstorage-mock';
import { entities } from './mockEntities';
const timeRegex = /\d{2}:\d{2}/g;
const dateRegex = /20(2[3-9]|[3-9][0-9])-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z/g;
const ENTITIES = JSON.parse(JSON.stringify(entities)  
  // purposely using 2022 here as the regex should work for 9 years ;)
  .replace(dateRegex, '2022-07-23T03:14:01.164Z')
  .replace(timeRegex, '01:01'));

localStorage.setItem('hassTokens', JSON.stringify({
  access_token: "FAKE",
  token_type:"Bearer",
  expires_in: Number.MAX_SAFE_INTEGER,
  hassUrl: "http://fake.com",
  clientId: "http://localhost:6006/",
  expires: Number.MAX_SAFE_INTEGER,
  refresh_token:"FAKE"
}));
// eslint-disable-next-line react-refresh/only-export-components
export const connection = {
  id: 'fake'
}
// eslint-disable-next-line react-refresh/only-export-components
export const mocked = {
  callService: jest.fn(),
  getAllEntities: jest.fn().mockImplementation(() => ENTITIES),
  async getAuth() {
    return {
      expired: false,
    }
  },
  async getConfig() {
    return {
      time_zone: 'Australia/Sydney'
    }
  },
  async createConnection() {
    return connection;
  },
  subscribeEntities: jest.fn().mockImplementation((_: null, subScribeFn: (entities: HassEntities) => void) => {
    subScribeFn(ENTITIES);
  })
}

jest.mock('home-assistant-js-websocket', () => mocked);

import { HassConnect } from '@hakit/core';

// eslint-disable-next-line react-refresh/only-export-components
export const onReady = jest.fn();
// Custom wrapper component for testing
export const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  return <HassConnect onReady={onReady} hassUrl="http://fake.com">
    {children}
  </HassConnect>
}