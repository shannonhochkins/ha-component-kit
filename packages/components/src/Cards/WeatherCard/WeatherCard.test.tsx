// important that these are imported first or the mock won't work.
import { TestWrapper, onReady } from "@mocks/mockConnection";
import { WeatherCard } from "@components";
import { render, waitFor } from "@testing-library/react";

describe("<WeatherCard />", () => {
  beforeEach(() => {
    onReady.mockClear();
  });
  // Test to check if the component renders correctly
  it("renders WeatherCard with title from entity", async () => {
    const { getByTestId } = render(
      <WeatherCard entity="weather.entity" data-testid="weather-card" />,
      {
        wrapper: TestWrapper,
      }
    );
    await waitFor(() => expect(onReady).toHaveBeenCalledTimes(1));
    const buttonElement = getByTestId("weather-card");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });

  it("should show without forecast and custom icon", async () => {
    const { getByTestId } = render(
      <WeatherCard
        entity="weather.entity"
        icon="mdi:cross"
        includeForecast={false}
        data-testid="weather-card"
      />,
      {
        wrapper: TestWrapper,
      }
    );
    await waitFor(() => expect(onReady).toHaveBeenCalledTimes(1));
    const buttonElement = getByTestId("weather-card");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });

  it("should show without current & custom temperature suffix", async () => {
    const { getByTestId } = render(
      <WeatherCard
        entity="weather.entity"
        includeCurrent={false}
        temperatureSuffix={"XXX"}
        data-testid="weather-card"
      />,
      {
        wrapper: TestWrapper,
      }
    );
    await waitFor(() => expect(onReady).toHaveBeenCalledTimes(1));
    const buttonElement = getByTestId("weather-card");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });
});
