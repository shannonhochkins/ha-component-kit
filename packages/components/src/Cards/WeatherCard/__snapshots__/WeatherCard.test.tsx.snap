// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherCard /> renders WeatherCard with title from entity 1`] = `
.emotion-0 {
  all: unset;
  padding: 1rem;
  position: relative;
  overflow: hidden;
  border-radius: 1rem;
  width: var(--ha-device-weather-card-width);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  cursor: default;
  background-color: var(--ha-primary-background);
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  -webkit-transition: all 0.2s cubic-bezier(0.06, 0.67, 0.37, 0.99);
  transition: all 0.2s cubic-bezier(0.06, 0.67, 0.37, 0.99);
  gap: 1rem;
}

.emotion-0:hover {
  background-color: var(--ha-primary-background-hover);
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  all: unset;
  font-size: 0.8rem;
  color: var(--ha-secondary-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-4 {
  all: unset;
  font-size: 1rem;
  color: var(--ha-primary-color);
  margin-top: 0.3rem;
  margin-left: 1.1rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 0.5rem;
}

.emotion-7 {
  color: var(--ha-primary-color);
}

.emotion-8 {
  color: var(--ha-primary-color);
  font-size: 0.8rem;
}

.emotion-9 {
  color: var(--ha-secondary-color);
  font-size: 0.75rem;
}

<div
  class="emotion-0"
  data-testid="weather-card"
>
  <div
    class="emotion-1"
  >
    <span />
    <div
      class="emotion-2"
    >
      <h4
        class="emotion-3"
      >
        <span />
        Forecast &lt;City&gt;
      </h4>
      <h4
        class="emotion-4"
      >
        17.6
        °
        , 
        Partlycloudy
      </h4>
    </div>
  </div>
  <div
    class="emotion-1"
    style="justify-content: space-between;"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        Tue
      </div>
      <span />
      <div
        class="emotion-8"
      >
        20.8
        °
      </div>
      <div
        class="emotion-9"
      >
        12.7
        °
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        Wed
      </div>
      <span />
      <div
        class="emotion-8"
      >
        16.2
        °
      </div>
      <div
        class="emotion-9"
      >
        8.2
        °
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        Thu
      </div>
      <span />
      <div
        class="emotion-8"
      >
        18.7
        °
      </div>
      <div
        class="emotion-9"
      >
        5
        °
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        Fri
      </div>
      <span />
      <div
        class="emotion-8"
      >
        14.4
        °
      </div>
      <div
        class="emotion-9"
      >
        10.3
        °
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        Sat
      </div>
      <span />
      <div
        class="emotion-8"
      >
        15.4
        °
      </div>
      <div
        class="emotion-9"
      >
        6.2
        °
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        Sun
      </div>
      <span />
      <div
        class="emotion-8"
      >
        12.5
        °
      </div>
      <div
        class="emotion-9"
      >
        6.7
        °
      </div>
    </div>
  </div>
</div>
`;

exports[`<WeatherCard /> should show without current & custom temperature suffix 1`] = `
.emotion-0 {
  all: unset;
  padding: 1rem;
  position: relative;
  overflow: hidden;
  border-radius: 1rem;
  width: var(--ha-device-weather-card-width);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  cursor: default;
  background-color: var(--ha-primary-background);
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  -webkit-transition: all 0.2s cubic-bezier(0.06, 0.67, 0.37, 0.99);
  transition: all 0.2s cubic-bezier(0.06, 0.67, 0.37, 0.99);
  gap: 1rem;
}

.emotion-0:hover {
  background-color: var(--ha-primary-background-hover);
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 0.5rem;
}

.emotion-3 {
  color: var(--ha-primary-color);
}

.emotion-4 {
  color: var(--ha-primary-color);
  font-size: 0.8rem;
}

.emotion-5 {
  color: var(--ha-secondary-color);
  font-size: 0.75rem;
}

<div
  class="emotion-0"
  data-testid="weather-card"
>
  <div
    class="emotion-1"
    style="justify-content: space-between;"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Tue
      </div>
      <span />
      <div
        class="emotion-4"
      >
        20.8
        XXX
      </div>
      <div
        class="emotion-5"
      >
        12.7
        XXX
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Wed
      </div>
      <span />
      <div
        class="emotion-4"
      >
        16.2
        XXX
      </div>
      <div
        class="emotion-5"
      >
        8.2
        XXX
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Thu
      </div>
      <span />
      <div
        class="emotion-4"
      >
        18.7
        XXX
      </div>
      <div
        class="emotion-5"
      >
        5
        XXX
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Fri
      </div>
      <span />
      <div
        class="emotion-4"
      >
        14.4
        XXX
      </div>
      <div
        class="emotion-5"
      >
        10.3
        XXX
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Sat
      </div>
      <span />
      <div
        class="emotion-4"
      >
        15.4
        XXX
      </div>
      <div
        class="emotion-5"
      >
        6.2
        XXX
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Sun
      </div>
      <span />
      <div
        class="emotion-4"
      >
        12.5
        XXX
      </div>
      <div
        class="emotion-5"
      >
        6.7
        XXX
      </div>
    </div>
  </div>
</div>
`;

exports[`<WeatherCard /> should show without forecast and custom icon 1`] = `
.emotion-0 {
  all: unset;
  padding: 1rem;
  position: relative;
  overflow: hidden;
  border-radius: 1rem;
  width: var(--ha-device-weather-card-width);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  cursor: default;
  background-color: var(--ha-primary-background);
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  -webkit-transition: all 0.2s cubic-bezier(0.06, 0.67, 0.37, 0.99);
  transition: all 0.2s cubic-bezier(0.06, 0.67, 0.37, 0.99);
  gap: 1rem;
}

.emotion-0:hover {
  background-color: var(--ha-primary-background-hover);
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  all: unset;
  font-size: 0.8rem;
  color: var(--ha-secondary-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-4 {
  all: unset;
  font-size: 1rem;
  color: var(--ha-primary-color);
  margin-top: 0.3rem;
  margin-left: 1.1rem;
}

<div
  class="emotion-0"
  data-testid="weather-card"
>
  <div
    class="emotion-1"
  >
    <span />
    <div
      class="emotion-2"
    >
      <h4
        class="emotion-3"
      >
        <span />
        Forecast &lt;City&gt;
      </h4>
      <h4
        class="emotion-4"
      >
        17.6
        °
        , 
        Partlycloudy
      </h4>
    </div>
  </div>
</div>
`;
